### ISSUE TITLE
Lag

### ISSUE THREAD

**Issue by costingheonea** [merdok/homebridge-webos-tv#498]
Hi,
Great plugin. Only one issue I have.
A few seconds lag between pressing a button in homekit and tv response. I this time, the TV shows an “unresponsive” status until the tv responses.
Is there a solution for this?
Thank you!

**Comment by merdok** [comment]
Hi,
there should not be any lag. The issue must be either somewhere in your local network or the device on which you run homebridge has not enough performance.

**Comment by costingheonea** [comment]
The device I am running Homebridge is a Mac Mini with M2, so definetly this is not the problem.
I dont know about the local network. All other accessories are running fine: cameras on scrypted, AC units that also run locally with their own IP.
What shoul I be looking for about the local network?

**Comment by merdok** [comment]
I cannot tell for what you have to look exactly as I never experienced such issue myself.
What I can tell is that the issue lies somewhere in your environment. 
I am not certain but I recall that there might have been a similar issue in the past before so you can also try to search the old issues.

**Comment by francescosabia** [comment]
I have this lag too.. after updating to iOS 17

**Comment by francescosabia** [comment]
> Same here, after updating to iOS 17, there's definitely something that's not working properly. All the other devices I manage through homebridge are just fine though. It' just the two LG tvs I use through this plugin that now are exhibiting this annoying behavior where it says "No response" then after a slight delay it will respond to your command, and it does that every time you open up the device menu on the Home App. Nothing changed on my side except for the update to iOS 17 on all of my  devices, HomePod and AppleTV included. Could you please take a look a this ? Thank you!

I agree. I also tried another webos plugin and it has the same problem. But I totally prefer this plugin (I don t understand why the dev doesn t update the older 2.4.1 instead to release a new plugin with 2.4.6). But I really hope there will be a fix for ios17!

**Comment by costingheonea** [comment]
Yes, I can confirm the issue is with the new iOS 17. And the reason why I am saying this is the lag is almost zero when using the Home app from my Macbook with macOS Ventura. I suppose the latest version, macOS Sonoma, will have the same issue like iOS 17.
I posted 2 videos where you can see the difference:
iOS 17: https://www.veed.io/view/1c9f2085-b4ca-46a9-8d48-1014fd6bc181?sharingWidget=true&panel=share
macOS Ventura: https://www.veed.io/view/7fe0f2ae-2005-4770-b670-3b1313f5d315?panel=share
I hope you can fix this soon. 
Thank you!

**Comment by merdok** [comment]
Well if this is a ios17 specific issue then I will certainly have a look at it, but it will take some time as I am currently traveling abroad...

**Comment by francescosabia** [comment]
> Yes, I can confirm the issue is with the new iOS 17. And the reason why I am saying this is the lag is almost zero when using the Home app from my Macbook with macOS Ventura. I suppose the latest version, macOS Sonoma, will have the same issue like iOS 17. I posted 2 videos where you can see the difference: iOS 17: https://www.veed.io/view/1c9f2085-b4ca-46a9-8d48-1014fd6bc181?sharingWidget=true&panel=share macOS Ventura: https://www.veed.io/view/7fe0f2ae-2005-4770-b670-3b1313f5d315?panel=share I hope you can fix this soon. Thank you!

Yes the new app has 0 lag is fantastic! I hope the plugin dev can fix it 🙏🏻🙏🏻

**Comment by costingheonea** [comment]
What new app are you talking about? Is there another plugin that has 0lag on iOS17?

**Comment by francescosabia** [comment]
> What new app are you talking about? Is there another plugin that has 0lag on iOS17?

Sorry I mean the home app in ios17! It has 0 lag when I open it.. very fast

**Comment by merdok** [comment]
So i had a look at this and it seems that it is actually an issue with the new Home app in iOS17. The new Home app simply cannot handle that much accessories in a single group, not sure why... The solution here would be to either reduce the amount of accessories (by removing unneeded switches in the plugin config) or simply break the group (by selecting "Show as separate tiles" in the group view in the Home app).

I could also introduce the possibility to add the same TV multiple times, that would allow to create separate groups with less switches (like a group with only channels, or only inputs, or maybe one with only remote controls). That could also help here...

**Comment by francescosabia** [comment]
The new home app is very fast but this bug (and also other like the slow streaming of the cameras when not at home) are very annoying.. i hope there will be other solutions for this plugin to work better.. thanks for the reply and keep the good work!

**Comment by francescosabia** [comment]
> So i had a look at this and it seems that it is actually an issue with the new Home app in iOS17. The new Home app simply cannot handle that much accessories in a single group, not sure why... The solution here would be to either reduce the amount of accessories (by removing unneeded switches in the plugin config) or simply break the group (by selecting "Show as separate tiles" in the group view in the Home app).
> 
> I could also introduce the possibility to add the same TV multiple times, that would allow to create separate groups with less switches (like a group with only channels, or only inputs, or maybe one with only remote controls). That could also help here...

I reduced the tv group from 21 to 11 accessories.. but still have the lag..

**Comment by merdok** [comment]
This issue has nothing to do with the plugin, the new Home app simply cannot handle much accessories in a group as it seems. It happens only in the fancy view with big switches, as soon as you navigate inside the single accessory there is no lag anymore. Also if you start the plugin with its default settings, where only a couple of switches are present, there is also no lag.

**Comment by francescosabia** [comment]
I can t have less than 9 accessories.. less than this I ll use the native integration 😂 I hope Apple will fix thisInviato da iPhoneIl giorno 24 set 2023, alle ore 20:22, Marcin ***@***.***> ha scritto:﻿
This issue has nothing to do with the plugin, the new Home app simply cannot handle much accessories in a group as it seems. It happens only in the fancy view with big switches, as soon as you navigate inside the single accessory there is no lag anymore. Also if you start the plugin with its default settings, where only a couple of switches are present, there is also no lag.

**Comment by costingheonea** [comment]
I can confirm that if I use as separate accessories, the lag is 0. I hope we will get a fix in iOS 17 by Apple or maybe by you because. I had lag only with 4 switches.

**Comment by merdok** [comment]
Every plugin will have the issue as long as the accessory group will have many accessories in it. You see the issue particularly with this plugin as there are usually a lot of accessories in the group here because of all the features.

Until Apple hopefully does something about it soon and improved the new Home app UI, a workaround would be to allow multiply accessories for the same tv, which is currently not possible, that would allow you to create dedicated groups with less accessories in it which should mitigate the issue.

**Comment by francescosabia** [comment]
> Every plugin will have the issue as long as the accessory group will have many accessories in it. You see the issue particularly with this plugin as there are usually a lot of accessories in the group here because of all the features.
> 
> Until Apple hopefully does something about it soon and improved the new Home app UI, a workaround would be to allow multiply accessories for the same tv, which is currently not possible, that would allow you to create dedicated groups with less accessories in it which should mitigate the issue.

I reduced to 8/9 accessories without luck, I have grouped other accessories (of other plugin) in 2/3 and they work good.. I doubt Apple will do something.. I hope at least they know

**Comment by francescosabia** [comment]
> This issue has nothing to do with the plugin, the new Home app simply cannot handle much accessories in a group as it seems. It happens only in the fancy view with big switches, as soon as you navigate inside the single accessory there is no lag anymore. Also if you start the plugin with its default settings, where only a couple of switches are present, there is also no lag.

This 2.4.6 version is yours? https://plugins.hoobs.org/plugin/@su-thomas/homebridge-webos-tv

I am using this now but with same bug

**Comment by merdok** [comment]
No, I did not release a 2.4.6 version. The latest is 2.4.1

**Comment by francescosabia** [comment]
> No, I did not release a 2.4.6 version. The latest is 2.4.1

But it s the same as your plugin.. he also mentioned you

**Comment by merdok** [comment]
Probably a fork, but I am not sure what the changes are.

**Comment by francescosabia** [comment]
> Probably a fork, but I am not sure what the changes are.

I hope there will be a fix for the lag.. in others groups of accessories I don t have the lag

**Comment by mikellas1983** [comment]
Just for everyones information. This happens to native accessories as well. Group accessories and others that contain a slider suffer from this bug. I have issues with IKEA , Aqara and other native integrated vendors.No issues whatsoever with a device that has iOS 16 on.

**Comment by francescosabia** [comment]
> Just for everyones information. This happens to native accessories as well. Group accessories and others that contain a slider suffer from this bug. I have issues with IKEA , Aqara and other native integrated vendors.No issues whatsoever with a controlling device that has iOS 16 on.

The bug appena also in groups without accessories that have sliders…I have the Tv group with 8 accessories without anyone with a slider and there is the bug.. I tried to do other groups and I dont have the bug

**Comment by mikellas1983** [comment]
This is an issue with HOME app and iOS 17 

https://www.reddit.com/r/HomeKit/comments/16ob1tu/ios17_sliders_very_slow/

**Comment by francescosabia** [comment]
> This is an issue with HOME app and iOS 17
> 
> https://www.reddit.com/r/HomeKit/comments/16ob1tu/ios17_sliders_very_slow/

Yes but I only have the bug with this plugin..

**Comment by merdok** [comment]
Like already mentioned it is a bug in the iOS17 Home app and it all depends on how many accessories and what kind of accessories you have in your group.
